critwall/src/intTest/java/org/opencadc/critwall/FileSyncJobTest.java

```diff
@@ -195,31 +212,111 @@ public void testValidJob() {
                 1008000L);
 
             log.debug("putting test artifact to database");
-            dao.put(artifactToUpdate);
+            dao.put(artifactToUpdate, true);
             // verify something was done
-            UUID artifactUUID = artifactToUpdate.getID();
+            artifactUUID = artifactToUpdate.getID();
             Assert.assertNotNull(artifactUUID);
-            Artifact testArtifact = dao.get(artifactUUID);
-            Assert.assertNotNull(testArtifact);
-
             log.debug("test artifact stored: UUID is " + artifactUUID.toString());
 
             FileSyncJob fsj = new FileSyncJob(artifactID, resourceID, sa, dao);
             fsj.run();
 
             // check job succeeded by trying to get artifact by location
             Artifact storedArtifact = dao.get(artifactID);
+            Assert.assertNotNull("storage location of artifact should not be null.", storedArtifact.storageLocation);
 
             // check for file on disk, throw away the bytes
             FileOutputStream dest = new FileOutputStream("/dev/null");
             log.debug("shunting to /dev/null");
             sa.get(storedArtifact.storageLocation, dest);
 
         } catch (Exception unexpected) {
-            Assert.fail("unexpected exception");
+            Assert.fail("unexpected exception: " + unexpected);
             log.debug(unexpected);
         }
         log.info("testValidJob - DONE");
     }
 
+    @Test
+    public void testInvalidJobBadChecksum() {
+        final String testDir = TEST_ROOT + File.separator + "testValidJobBadChecksum";
+        UUID artifactUUID = null;
+
+        try {
+            createTestDirectory(testDir);
+
+            OpaqueFileSystemStorageAdapter sa = new OpaqueFileSystemStorageAdapter(new File(TEST_ROOT), 1);
+
+            URI artifactID = new URI(TEST_ARTIFACT_URI);
+            URI resourceID = new URI(TEST_RESOURCE_ID);
+
+            // Set up an Artifact in the database to start.
+            // Set checksum to the wrong value.
+            Artifact artifactToUpdate = new Artifact(artifactID,
```

Comment:
Can you make the bad checksum more obviously incorrect here? When I looked, it started the same as the correct one and it took some time to see the discrepancy. Instead of:
```
md5:646d3c548ffb98244a0fc52b6055555
```
maybe this looks more made-up:
```
md5:0123456789abcdef0123456789abcdef
```
note:  used sequence so it's easier to get the right length

Reply:
ok